configs/gateway/nginx-http.conf

```diff
@@ -19,6 +19,10 @@ server {
     add_header Referrer-Policy strict-origin-when-cross-origin always;
     add_header Strict-Transport-Security max-age=31536000 always;
 
+    # disable request entity max size.
+    # http://nginx.org/en/docs/http/ngx_http_core_module.html#client_max_body_size
+    client_max_body_size 0;
```

Comment:
Not sure what else can be done if we don't want to 'unlimit' since there's no way we can predict how big the file can be

Reply:
I *think* `0` is the default in GitLab (https://gitlab.com/gitlab-org/omnibus-gitlab/-/blob/2684ff46c0c8acec9fb5bd5c5eeec5b8c68e49d7/files/gitlab-cookbooks/gitlab/attributes/default.rb#L666)

It should be OK so long as the application validates input data. For example, in 2013 the Django framework suffered DoS attacks whereby people put large passwords and the hashing algorithm took forever to run. Their fix was to restrict `client_max_body_size`.

We could set it to some arbitrary size like `100M` but that too could suffer the same issue as described above.